scripts/exgfs_wave_prep.sh

```diff
@@ -68,30 +65,31 @@ EOF
   # Set time stamps for model start and output
   # For special case when IAU is on but this is an initial half cycle
   if [[ ${IAU_OFFSET} -eq 0 ]]; then
-    ymdh_beg=${YMDH}
+    ymdh_beg=${PDY}${cyc}
   else
-    ymdh_beg=$(${NDATE} -${WAVHINDH} ${YMDH})
+    ymdh_beg=$(date --utc +%Y%m%d%H -d "${PDY} ${cyc} - ${WAVHINDH} hours")
   fi
   time_beg="$(echo ${ymdh_beg} | cut -c1-8) $(echo ${ymdh_beg} | cut -c9-10)0000"
```

Comment:
Casting the date string in the desired format using `date` instead of `cut`
```suggestion
  time_beg=$(date --utc -d "${ymdh_beg:0:8} ${ymdh_beg:8:2}" +"%Y%m%d %H0000")
```